presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -480,7 +496,22 @@ public boolean mkdirs(Path f, FsPermission permission)
         return true;
     }
 
-    private Iterator<LocatedFileStatus> listPrefix(Path path)
+    /**
+     * Enum representing the valid listing modes. This could be two booleans (recursive, filesOnly) except
```

Comment:
I feel filesOnly in the comment a little confusing. what do you think? Maybe just keep:
`Enum representing the valid listing modes`
as comment, or we do not need the comment at all?